lib/browser/mqtt.ts

```diff
@@ -383,6 +430,25 @@ export class MqttClientConnection extends BufferedEventEmitter {
         });
     }
 
+    /**
+     * Returns seconds until next reconnect attempt.
+     */
+    private get_reconnect_time_sec(): number {
+        if (this.reconnect_min_sec == 0 && this.reconnect_max_sec == 0) {
+            return 0;
+        }
+
+        // Uses "FullJitter" backoff algorithm, described here:
+        // https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
+        // We slightly vary the algorithm described on the page,
+        // which takes (base,cap) and may result in 0.
+        // But we take (min,max) as parameters, and don't don't allow results less than min.
+        const cap = this.reconnect_max_sec - this.reconnect_min_sec;
+        const base = Math.max(this.reconnect_min_sec, 1);
+        const sleep = Math.random() * Math.min(cap, base * (2 ** this.reconnect_count));
```

Comment:
Is JS math unbounded?  If reconnect_count gets large due to extended connectivity failures, what happens?

Reply:
If the number is out of bound, JS will return definition "INFINITY" which will be treat as a infinite large number here. The function will work fine. 
Change ** operator to Math.pow though, since IE does not support ** operator.